tests/Integration/Behaviour/Features/Context/Domain/Product/UpdatePackFeatureContext.php

```diff
@@ -108,15 +114,71 @@ public function assertPackEmpty(string $packReference): void
      * @param string $packReference
      * @param TableNode $table
      */
-    public function assertPackContents(string $packReference, TableNode $table): void
+    public function assertPackContentsForDefaultShop(string $packReference, TableNode $table): void
+    {
+        $this->assertPackContents($packReference, $table, ShopConstraint::shop($this->getDefaultShopId()));
+    }
+
+    /**
+     * @Then pack :packReference should contain products with following details for shop(s) :shopReferences:
+     *
+     * @param string $packReference
+     * @param TableNode $table
+     * @param string $shopReferences
+     */
+    public function assertPackContentsForShops(string $packReference, TableNode $table, string $shopReferences): void
+    {
+        foreach (explode(',', $shopReferences) as $shopReference) {
```

Comment:
We should start reusing this. Method in some abstract context should be enough. Or if we want to go even further maybe we could think of some reusable transformation. Anyway, just a note, I think such refacto could be done in separate pr for multiple contexts at a time.